tests/lib/fortigate_test.py

```diff
@@ -0,0 +1,390 @@
+                         self.naming), EXP_INFO
+    )
+
+    any_src_acl = fortigate.Fortigate(
+      policy.ParsePolicy(GOOD_HEADER + any_src_term,
+                         self.naming), EXP_INFO
+    )
+
+    any_dest_acl = fortigate.Fortigate(
+      policy.ParsePolicy(GOOD_HEADER + any_dest_term,
+                         self.naming), EXP_INFO
+    )
+
+    src_sig = 'set srcaddr 10.0.0.0/8'
+    dest_sig = 'set dstaddr 20.0.0.0/8'
+    any_dest_sig = 'set dstaddr all'
+    any_src_sig = 'set srcaddr all'
+
+    self.assertTrue(
+      src_sig in str(diff_addr_acl) and dest_sig in str(diff_addr_acl),
+      '[%s]' % str(diff_addr_acl)
+    )
+    # [] check acl generate one 'set subnet' for dup addresses
+    self.assertEqual(
+      str(same_addr_acl).count('set subnet'), 1
+    )
+    self.assertTrue(
+      any_src_sig in str(any_src_acl), '[%s]' % str(any_src_acl)
+    )
+    self.assertTrue(
+      any_dest_sig in str(any_dest_acl), '[%s]' % str(any_dest_acl)
+    )
+
+  def testServices(self):
+    dup_port_term = self.fmt.format(TERM_TEMPLATE,
+                                    src_port='HTTP',
+                                    dest_port='HTTP')
+    diff_port_term = self.fmt.format(TERM_TEMPLATE,
+                                     src_port='HTTP',
+                                     dest_port='HTTPS')
+    src_only_term = self.fmt.format(TERM_TEMPLATE,
+                                    src_port='HTTP',
+                                    remove_fields=('dest_port',))
+    icmp_term = self.fmt.format(TERM_TEMPLATE,
+                                protocol='icmp',
+                                remove_fields=('dest_port', 'src_port'))
+    ip_term = self.fmt.format(TERM_TEMPLATE,
+                              remove_fields=('dest_port', 'src_port', 'protocol'))
+    custom_port_term = self.fmt.format(TERM_TEMPLATE, src_port='WHOIS')
+
+    dup_acl = fortigate.Fortigate(policy.ParsePolicy(
+      GOOD_HEADER + dup_port_term,
+      self.naming), EXP_INFO
+    )
+    diff_acl = fortigate.Fortigate(policy.ParsePolicy(
+      GOOD_HEADER + diff_port_term,
+      self.naming), EXP_INFO
+    )
+    src_only_acl = fortigate.Fortigate(policy.ParsePolicy(
+      GOOD_HEADER + src_only_term,
+      self.naming), EXP_INFO
+    )
+    icmp_acl = fortigate.Fortigate(policy.ParsePolicy(
+      GOOD_HEADER + icmp_term,
+      self.naming), EXP_INFO
+    )
+    ip_acl = fortigate.Fortigate(policy.ParsePolicy(
+      GOOD_HEADER + ip_term,
+      self.naming), EXP_INFO
+    )
+    custom_port_acl = fortigate.Fortigate(
+      policy.ParsePolicy(GOOD_HEADER + custom_port_term,
+                         self.naming), EXP_INFO
+    )
+
+    dup_sig = 'set service HTTP\n'
+    diff_sig = 'set service HTTP HTTPS\n'
+    src_only_sig = dup_sig
+    icmp_sig = 'set service ALL_ICMP\n'
+    ip_sig = 'set service ALL\n'
+    custom_port_sig = 'config firewall service custom\n\tedit 43\n\t\t' \
```

Comment:
do not use \ for line continuation.
http://google.github.io/styleguide/pyguide.html#32-line-length
The link provides ways to get around this.